File: javascript/algorithms/bubbleSort.js
function bubbleSort(values) {
  for (let i = values.length - 1; i >= 0; i--) {
    for(var j = 1; j <= i; j++) {
      if(values[j - 1] > values[j]) {
           let temporary = values[j - 1];
           values[j - 1] = values[j];
           values[j] = temporary;
        }
     }
   }
   return values;
}

console.log(bubbleSort([93, 2, 82, 44, 210, 1, 39])); 

File: erlang/algorithms/bubble-sort.erl
-module(bubblesort).
-export([sort/1]).
-import(lists, [reverse/1]).

sort(L) ->
  sort(L, [], true).

sort([], L, true) ->
  reverse(L);
  
sort([], L, false) ->
  sort(reverse(L), [], true);
  
sort([ X, Y | T ], L, _) when X > Y ->
  sort([ X | T ], [ Y | L ], false);
  
sort([ X | T ], L, Halt) ->
sort(T, [ X | L ], Halt).
